modules/os2forms_fasit/src/Helper/CertificateLocatorHelper.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace Drupal\os2forms_fasit\Helper;
+
+use Drupal\os2forms_fasit\Exception\CertificateLocatorException;
+use GuzzleHttp\Client;
+use Http\Adapter\Guzzle7\Client as GuzzleAdapter;
+use Http\Factory\Guzzle\RequestFactory;
+use ItkDev\AzureKeyVault\Authorisation\VaultToken;
+use ItkDev\AzureKeyVault\KeyVault\VaultSecret;
+use ItkDev\Serviceplatformen\Certificate\AzureKeyVaultCertificateLocator;
+use ItkDev\Serviceplatformen\Certificate\CertificateLocatorInterface;
+use ItkDev\Serviceplatformen\Certificate\FilesystemCertificateLocator;
+
+/**
+ * Certificate locator helper.
+ */
+class CertificateLocatorHelper {
+  public const LOCATOR_TYPE = 'locator_type';
+  public const LOCATOR_TYPE_AZURE_KEY_VAULT = 'azure_key_vault';
+  public const LOCATOR_TYPE_FILE_SYSTEM = 'file_system';
+  public const LOCATOR_PASSPHRASE = 'passphrase';
+  public const LOCATOR_AZURE_KEY_VAULT_TENANT_ID = 'tenant_id';
+  public const LOCATOR_AZURE_KEY_VAULT_APPLICATION_ID = 'application_id';
+  public const LOCATOR_AZURE_KEY_VAULT_CLIENT_SECRET = 'client_secret';
+  public const LOCATOR_AZURE_KEY_VAULT_NAME = 'name';
+  public const LOCATOR_AZURE_KEY_VAULT_SECRET = 'secret';
+  public const LOCATOR_AZURE_KEY_VAULT_VERSION = 'version';
+  public const LOCATOR_FILE_SYSTEM_PATH = 'path';
+
+  /**
+   * Constructor.
+   */
+  public function __construct(private readonly Settings $settings) {
+  }
+
+  /**
+   * Get certificate locator.
+   */
+  public function getCertificateLocator(): CertificateLocatorInterface {
+    $certificateSettings = $this->settings->getCertificate();
+
+    $locatorType = $certificateSettings[self::LOCATOR_TYPE];
+    $options = $certificateSettings[$locatorType];
+    $options += [
+      self::LOCATOR_PASSPHRASE => $certificateSettings[self::LOCATOR_PASSPHRASE] ?: '',
+    ];
+
+    if (self::LOCATOR_TYPE_AZURE_KEY_VAULT === $locatorType) {
+      $httpClient = new GuzzleAdapter(new Client());
+      $requestFactory = new RequestFactory();
```

Comment:
Would one not inject a Drupal::httpClient using PSR18/17 interfaces and not be locked into guzzle?

Reply:
PSR18/17 does not have built-in mechanisms for working with certificates. This means we would end up having a hard requirement on some sort of adapter (e.g. Guzzle) in order to end up with a PSR18 compliant client.